File: data-client/src/main/java/org/wikipedia/dataclient/mwapi/page/MwThankPostResponse.java
package org.wikipedia.dataclient.mwapi.page;


import org.wikipedia.dataclient.mwapi.MwResponse;

public class MwThankPostResponse extends MwResponse {
  private Result result;

  public Result getResult() {
    return result;
  }

  public void setResult(Result result) {
    this.result = result;
  }

  public class Result {
    private Integer success;
    private String recipient;

    public Integer getSuccess() {
      return success;
    }

    public void setSuccess(Integer success) {
      this.success = success;
    }

    public String getRecipient() {
      return recipient;
    }

    public void setRecipient(String recipient) {
      this.recipient = recipient;
    }

  }
}

File: app/src/main/java/fr/free/nrw/commons/actions/MwThankPostResponse.kt
package fr.free.nrw.commons.actions

import org.wikipedia.dataclient.mwapi.MwResponse

/**
 * Response of the Thanks API.
 * Context:
 * The Commons Android app lets you thank other contributors who have uploaded a great picture.
 * See https://www.mediawiki.org/wiki/Extension:Thanks
 */
class MwThankPostResponse : MwResponse() {
    var result: Result? = null

    inner class Result {
        var success: Int? = null
        var recipient: String? = null
    }
}
